ios/RNSScreen.mm

```diff
@@ -902,6 +902,13 @@ - (void)updateFormSheetPresentationStyle
           [self setAllowedDetentsForSheet:sheet
                                        to:[self detentsFromMaxHeightFractions:_sheetAllowedDetents]
                                   animate:NO];
+
+          if (_sheetInitialDetent > 0 && _sheetInitialDetent < _sheetAllowedDetents.count) {
+            UISheetPresentationControllerDetent *detent = sheet.detents[_sheetInitialDetent];
+            [self setSelectedDetentForSheet:sheet to:detent.identifier animate:YES];
+          } else if (_sheetInitialDetent != 0) {
+            RCTLogError(@"[RNScreens] sheetInitialDetent out of bounds for sheetAllowedDetents array");
+          }
```

Comment:
Maybe one thing that strikes me, what will happen on iOS 15? This code won't execute right and we still want to allow for choice there. Can we make is somehow work?